docs/lib/restapi/fragments/native_common/authz/common.md

```diff
@@ -0,0 +1,124 @@
+
+Your requests will be automatically have the IAM credentials when your API is set up to use IAM as the authorization mode. For example, you can check this by looking at what authorization type is configured for the API
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "IAM"
+        }
+    }
+}
+```
+
+## API Key
+
+If you want to configure the REST API to be a public API, you can set an API key on the API Gateway and update your configuration to use the API key. You should have the following configuration in your `amplifyconfiguration.json` file containing the `API_KEY` authorization mode and the value of the API key to be used when creating requests.
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "API_KEY",
+            "API_KEY": "[API_KEY]"
+        }
+    }
+}
+```
+
+
+## Cognito User pool authorization
+
+If you set up the API Gateway with a custom authorization with a Cognito User pool, then you can set the `amplifyconfiguration.json` to use `COGNITO_USER_POOL`. 
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "COGNITO_USER_POOL"
+        }
+    }
+}
+```
+
+Also in `amplifyconfiguration.json` file, under `awsCognitoAuthPlugin` will contain the corresponding Cognito configuration values that refer to the Cognito User Pool you want to use.
+```json
+{
+    "CognitoUserPool": {
+        "Default": {
+            "PoolId": "POOL-ID",
+            "AppClientId": "APP-CLIENT-ID",
+            "AppClientSecret": "APP-CLIENT-SECRET",
+            "Region": "us-east-1"
+        }
+    },
+    "CredentialsProvider": {
+        "CognitoIdentity": {
+            "Default": {
+                "PoolId": "YOUR-COGNITO-IDENTITY-POOLID",
+                "Region": "us-east-1"
+            }
+        }
+    }
+}
+```
+
+This will automatically add the Authorization header with access token to your requests to this API. To see more details on how to configure the API Gateway with the custom authorization, see [this](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-integrate-with-cognito.html)
+
+
+## Custom Interceptor
+
+If you have a custom token from another provider, you can set it `custom` and create your own interceptor
+
+```
+CUSTOM INTERCEPTOR CODE
+```
+
```

Comment:
I'm not sure about this one. Where does the user set `custom`? Does this work on Android? Should we just skip this section for now? We can always come back and add it later. Or is the `CUSTOM INTERCEPTOR CODE` mean to be in Swift/Kotlin code?

Reply:
that must have been WIP text, i've updated in regards to using `OPENID_CONNECT` in amplifyconfiguration.json, and added the corresponding files